compysition/event.py

```diff
@@ -159,14 +157,32 @@ def event_id(self, event_id):
         else:
             self._event_id = event_id
 
+    def _list_get(self, obj, key, default):
+        try:
+            return obj[int(key)]
+        except (ValueError, IndexError, TypeError, KeyError, AttributeError) as e:
+            return default
+
+    def _getattr(self, obj, key, default):
+        try:
+            return getattr(obj, key, default)
+        except TypeError as e:
+            return default
+
+    def _obj_get(self, obj, key, default):
+        try:
+            return obj.get(key, default)
+        except (TypeError, AttributeError) as e:
+            return default
+
     def lookup(self, path):
         """
-        TODO: Account for list objects in the lookup path and generate multiple results if found
+        Implements the retrieval of a single list index through an integer path entry
         """
         if isinstance(path, str):
             path = [path]
 
-        value = reduce(lambda obj, key: obj.get(key, NullLookupValue()) if isinstance(obj, dict) else getattr(obj, key, NullLookupValue()), [self] + path)
+        value = reduce(lambda obj, key: self._obj_get(obj, key, self._getattr(obj, key, self._list_get(obj, key, NullLookupValue()))), [self] + path)
```

Comment:
Can you fill me in on why this was changed?

Reply:
To implement the retrieval of a list item and to limit the use of isinstance.  Also, I guess I didn't catch it before but, eventually I would like to replace the NullLookupValue with None.